endpoint.ts

```diff
@@ -128,39 +128,55 @@ function errorResponse(err: Error) {
 
 const getErrorSvg = (err: string) => {
   const splitMessage = (msg: string): string[] => {
+    msg = msg.includes(":") ? msg.replace(/[^:]+:/, "") : msg
     const chunks: string[] = []
     let currentChunk = ""
-
     msg.split(" ").forEach((word) => {
-      if ((currentChunk + word).length > 32) {
+      if ((currentChunk + word).length > 33) {
         chunks.push(currentChunk.trim())
         currentChunk = ""
       }
       currentChunk += `${word} `
     })
     chunks.push(currentChunk.trim())
-    return chunks.slice(0, 3) // Max 3 lines
+    return chunks
   }
 
   const errorLines = splitMessage(err)
+  const lineHeight = 24
+  const topMargin = 50
+  const bottomMargin = 20
+  const horizontalPadding = 40
+  const estimatedCharWidth = 8
+  const maxLineLength = errorLines.reduce(
+    (max, line) => Math.max(max, line.length),
+    0,
+  )
+  const dynamicWidth = maxLineLength * estimatedCharWidth + horizontalPadding
+  const dynamicHeight =
+    topMargin + errorLines.length * lineHeight + bottomMargin
+  const desiredRatio = 16 / 9
+  let finalWidth = dynamicWidth
+  let finalHeight = dynamicHeight
+  if (dynamicWidth / dynamicHeight < desiredRatio) {
+    finalWidth = dynamicHeight * desiredRatio
+  } else if (dynamicWidth / dynamicHeight > desiredRatio) {
+    finalHeight = dynamicWidth / desiredRatio
+  }
+  const viewBox = `0 0 ${finalWidth} ${finalHeight}`
```

Comment:
this code is getting a bit complex, can we move it to a separate file e.g. `getErrorSvg.ts`?